### PR TITLE
[REF] web,*: remove rpc service (it is now a function)

### PR DESCRIPTION
**Pull request by aab-odoo**
This PR aims at simplifying the way developpers can do rpcs
with our framework. Before this commit, in components, they had to
import the hook `useService`, and in the setup do something like
`this.rpc = useService("rpc")`, and then use `this.rpc` where they
wanted. In services, they had to explicitely declare the dependency
to the rpc service. In any other functions (like helper functions),
it was a bit painful: the function had to take an env (or the rpc
function directly) in arguments, which wasn't ideal.

This PR makes doing rpcs easier. The rpc service service has
been removed. The `jsonrpc` function has been renamed into `rpc`
(and the file `rpc_service` into `rpc`). One can now simply do
`import { rpc } from "@web/core/network/rpc"` and use `rpc` where
he wants.

Note that this change has a (maybe) not insignificant side-effect.
We no longer have the "protection" on rpc. Before, when a component
used `useService("rpc")`, the returned function was tied to that
component instance. If the component was destroyed when calling the
function, it crashed (s.t. a destroyed component didn't do rpc). If
the component was destroyed during the rpc, the promise was left
pending forever (s.t. subsequent callbacks weren't executed). This
protection was sometimes annoying, and people started to bypass it
by directly calling `env.services.rpc` (i.e. a version which isn't
tied to the component). We believe this protection is not really
necessary. In the future, we may introduce an `alive` function to
wrap rpc calls, and use it at some places where it is indeed
necessary (i.e. move it user-space instead of enforced by the
framework).

Part of task~3621046

### CODE REVIEW

**Review by ged-odoo — APPROVED** [review]
looks good to me

File: addons/im_livechat/static/src/embed/common/livechat_service.js [hunk 1]
```diff
@@ -285,14 +284,7 @@ export class LivechatService {
...
 export const livechatService = {
-    dependencies: [
-        "bus_service",
-        "mail.chat_window",
-        "mail.store",
-        "notification",
-        "notification",
-        "rpc",
-    ],
+    dependencies: ["bus_service", "mail.chat_window", "mail.store", "notification", "notification"],
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/embed/common/livechat_service.js:287]: It was already before the diff, but `notification` is there twice.

File: addons/im_livechat/static/src/embed/cors/boot.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ import { session } from "@web/session";
         {
             async: true,
             start(env) {
-                return function rpc(route, params = {}, settings) {
+                return function _rpc(route, params = {}, settings) {
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/embed/cors/boot.js:26]: That won't work: it's a patch of the rpc service in a specific context.

File: addons/mail/static/tests/helpers/webclient_setup.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 /* @odoo-module */
 
+import { rpc } from "@web/core/network/rpc";
```
Comment (seb-odoo) [inline addons/mail/static/tests/helpers/webclient_setup.js:3]: (minor) ideally sorted by path with the other imports from web below

**Review by seb-odoo — CHANGES_REQUESTED** [review]

File: addons/im_livechat/static/src/embed/cors/boot.js (hunk 2 above)
Comment (aab-odoo) [reply addons/im_livechat/static/src/embed/cors/boot.js:26]: oups indeed, I'll fix it

**Review by AntoineVDV — APPROVED** [review]

**Review by arj-odoo — APPROVED** [review]

**Comment by sdegueldre** [comment]
@robodoo r+

**Pull request closed without merging**